log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala

```diff
@@ -118,4 +120,28 @@ class SolrCloudStreamTest extends FunSuite with BaseSolrCloudTest {
     assert(streamFailed)
   }
 
+  test("should flush after n events") {
+    val stream = new TestStream(1, 10)
+
+    (1 to 10).foreach(x => stream.put("testApp", x.toString))
+    Thread.sleep(2500)
+    assert(stream.results.length === 10)
+  }
+
+  test("should flush after n seconds") {
+    val stream = new TestStream(1, 10)
+
+    (1 to 3).foreach(x => stream.put("testApp", x.toString))
+    Thread.sleep(2500)
+    assert(stream.results.length === 3)
+  }
+}
+
+class TestStream(flushDuration: Int, flushSize: Int) extends Stream[String](StreamParams()) {
+  StreamParams.apply(1, 100, flushSize, flushDuration, "fail")
```

Comment:
No need to call `apply` here, you can just call `StreamParams(args)`

Reply:
I will do this.